pkg/agent/proxy/proxier.go

```diff
@@ -516,6 +516,21 @@ func (p *proxier) installServices() {
 			nGroupID := groupID
 			if svcInfo.NodeLocalExternal() {
 				nGroupID, _ = p.groupCounter.Get(svcPortName, true)
+				var localEndpointUpdateList []k8sproxy.Endpoint
+				// If externalTrafficPolicy of the previous Service is Cluster, a group which only has local Endpoints
+				// should be installed.
+				if pSvcInfo != nil && !pSvcInfo.NodeLocalExternal() {
```

Comment:
There might be 3 issues:
1. If a Service's policy is local from the begining, the above code won't create a local group, adding code here will lead to inconsistent behavior.
2. Code is duplicated with the block under `needUpdateEndpoints`.
3. There is no cleanup when policy is changed from local to cluster.

I think we could just set `needUpdateEndpoints` to true if policy changes, just like when SessionAffinityType changes, then the existing code can take care of installing the local group.
For cleanup, it should uninstall the local group when NodeLocalExternal is false and local group id is found.

Reply:
I agree with the idea that just setting `needUpdateEndpoints` to true if policy changes, and this will be easier to understand. We also need a cleanup when policy is from Local to Cluster.